src/fmt.zig

```diff
@@ -42,6 +44,73 @@ pub fn resetWith(fmt: *Formatter, ast: IR) void {
     fmt.ast = ast;
 }
 
+/// Formats all roc files in the specified path.
+/// Handles both single files and directories
+/// Returns the number of files formatted.
+pub fn formatPath(gpa: std.mem.Allocator, base_dir: std.fs.Dir, path: []const u8) !usize {
+    var count: usize = 0;
+    // First try as a directory.
+    if (base_dir.openDir(path, .{ .iterate = true })) |const_dir| {
+        var dir = const_dir;
+        defer dir.close();
+        // Walk is recursive.
+        var walker = try dir.walk(gpa);
+        defer walker.deinit();
+        while (try walker.next()) |entry| {
+            if (entry.kind == .file) {
+                if (try formatFilePath(gpa, entry.dir, entry.basename)) {
+                    count += 1;
+                }
+            }
+        }
+    } else |_| {
+        if (try formatFilePath(gpa, base_dir, path)) {
+            count += 1;
+        }
+    }
+
+    return count;
+}
+
+fn formatFilePath(gpa: std.mem.Allocator, base_dir: std.fs.Dir, path: []const u8) !bool {
+    // Format any ".roc" files.
+    if (std.mem.eql(u8, std.fs.path.extension(path), ".roc")) {
```

Comment:
This would be better served as a guard statement, I think, to avoid unnecessary nesting.

Reply:
Very true.